.eslintrc.js

```diff
@@ -32,4 +32,58 @@ module.exports = {
     '@next/next/no-img-element': ['off'],
     'linebreak-style': ['error', os.platform() === 'win32' ? 'windows' : 'unix'],
   },
+  overrides: [
+    {
+      files: ['**/*.ts', '**/*.tsx'],
+      env: {
+        es2021: true,
+        node: true,
+        jest: true,
+      },
+      extends: [
+        'next',
+        'airbnb',
+        'airbnb-typescript',
+        'plugin:cypress/recommended',
+        'eslint:recommended',
+        'plugin:@typescript-eslint/eslint-recommended',
+        'plugin:@typescript-eslint/recommended',
+      ],
+      parser: '@typescript-eslint/parser',
+      parserOptions: {
+        ecmaVersion: 12,
+        sourceType: 'module',
+        project: ['./tsconfig.json'],
+      },
+      plugins: ['@typescript-eslint', 'eslint-plugin-jsx-a11y', 'cypress'],
+      rules: {
+        'function-call-argument-newline': ['off'],
+        'function-paren-newline': ['off'],
+        'import/prefer-default-export': ['off'],
+        'max-len': ['error', { code: 120 }],
+        'newline-per-chained-call': ['off'],
+        'no-console': ['off'],
+        'no-nested-ternary': ['off'],
+        'no-promise-executor-return': ['off'],
+        'no-underscore-dangle': ['off'],
+        semi: ['error', 'always'],
+        'react/jsx-filename-extension': ['off'],
+        'react/function-component-definition': ['off'],
+        'react/jsx-props-no-spreading': ['off'],
+        'react/no-danger': ['off'],
+        '@next/next/no-html-link-for-pages': ['off'],
+        '@next/next/no-img-element': ['off'],
+        'linebreak-style': ['error', os.platform() === 'win32' ? 'windows' : 'unix'],
+        '@typescript-eslint/no-unused-vars': [
+          'warn',
+          {
+            varsIgnorePattern: '^_',
+            argsIgnorePattern: '^_',
+          },
+        ],
```

Comment:
why not place all these rules in the rules section above? what's the purpose of the overrides section?

Reply:
> why not place all these rules in the rules section above? what's the purpose of the overrides section?

I used that because most of the project is still common js, so there'd be conflicts if I don't use separate rules for the linting. We'd just end up having erros everywhere.